tests/test_quit.py

```diff
@@ -1,5 +1,131 @@
-from quit import dummy
+import numpy as np
+import pytest
+from scipy import linalg
 
+from quit.basicfunction import (
+    basis,
+    bell_state,
+    bell_state_density,
+    bra,
+    braket,
+    dagger,
+    ket,
+    ketbra,
+    proj,
+    res,
+    unres,
+    unvec,
+    vec,
+)
 
-def test_dummy():
-    assert dummy() == 2137
+
+@pytest.mark.parametrize("phi", [np.pi, np.sqrt(2), 2j, 1 + 1j, 1])
+def test_ket_with_different_types_of_entries(phi):
+    np.testing.assert_array_equal(np.array([[1.0], [phi]]), ket([1, phi]))
+
+
+@pytest.mark.parametrize("dim", [4, 8])
+@pytest.mark.parametrize("index", [2, 3])
+def test_if_basis_is_correctly_defined(dim, index):
+    np.testing.assert_array_equal(basis(dim, index), np.identity(dim)[index])
+
+
+@pytest.mark.parametrize("phi", [np.pi, np.sqrt(2), 2j, 1 + 1j, 1])
+def test_bra_with_different_types_of_entries(phi):
+    np.testing.assert_array_equal(np.array([[1, np.conjugate(phi)]]), bra([1, phi]))
+
+
+@pytest.mark.parametrize("phi", [[np.sqrt(2), 1j + 2]])
+@pytest.mark.parametrize("psi", [[np.pi, 1, 2j]])
+def test_ketbra_is_equal_outer_product(phi, psi):
+    np.testing.assert_almost_equal(ketbra(phi, psi), np.outer(phi, np.conjugate(psi)))
+
+
+@pytest.mark.parametrize("phi", [[np.sqrt(2), 1j + 2]])
+@pytest.mark.parametrize("psi", [[np.pi, 2j]])
+def test_braket(phi, psi):
+    np.testing.assert_almost_equal(braket(phi, psi), np.inner(np.conjugate(phi), psi))
+
+
+@pytest.mark.parametrize("vector", [[1, 2], [1, -1]])
+def test_projector_is_equal_proj_to_the_power_of_2(vector):
+    np.testing.assert_almost_equal(proj(vector) @ proj(vector), proj(vector))
+
+
+sx = np.array([[0, 1], [1, 0]])
+
+
+@pytest.mark.parametrize(
+    "symmatrix", [np.identity(3), np.asarray(linalg.hadamard(2, dtype=complex)), np.kron(sx, sx)]
+)
+def test_dagger_for_hermitian_matrices(symmatrix):
+    np.testing.assert_array_equal(dagger(symmatrix), symmatrix)
```

Comment:
The name of the test does not tell what is actually tested there. Is the dagger of the hermitian matrix the same? Different? If different then how?

I know that this case is trivial. However, we should always try to reinforce a good habit of writing meaningful test names. Please try to come up with a better name.

The same goes for the test cases below.